lib/html.rb

```diff
@@ -26,19 +26,26 @@ def self.run_server(system_description_store, opts, &block)
       Server.set :public_folder, File.join(Machinery::ROOT, "html")
       Server.set :static_cache_control, "no-cache"
 
-      if opts[:ip] != "localhost" && opts[:ip] != "127.0.0.1"
+      if opts[:public]
```

Comment:
As described I'd catch the case that both `public` and an `ip` is given and raise an exception because it's not clear what the intended behavior is in that case.

I also think the if structure could be flattened to something like

```ruby
if opts[:public] && opts[:ip]
  raise ...
elsif opts[:public]
  ...
elsif opts[:ip] == "0.0.0.0"
  ...
elsif opts[:ip] != "localhost" && opts[:ip] != "127.0.0.1"
  ...
end
```